src/Diagnostics.DataProviders/Utility/CertificateHelper.cs

```diff
@@ -18,31 +18,31 @@ internal static X509Certificate2 FindAndValidateCertificate(string certificateTh
             var cert = FindX509Certificate(certificateThumbprint, certificateStoreLocation);
             if (!cert.HasPrivateKey)
             {
-                throw new MetricsClientException(string.Format("Cert with Thumbprint [{0}] doesn't have a private key", cert.Thumbprint));
+                throw new InvalidOperationException(string.Format("Cert with Thumbprint [{0}] doesn't have a private key", cert.Thumbprint));
```

Comment:
This would not be considered an InvalidOperationException. You can just throw `Exception`. Please revisit the kind of exceptions you are throwing for this class. Either throw just a regular `Exception` type or find the appropriate type.

Reply:
I'd like to throw a regular `Exception` type exception.